### PR TITLE
Fixed adding a bike

### PR DESCRIPTION
**Pull request by betsy-tse1**
Adding a bike as an admin will now work; the list will populate on the bikes_list page. Adding as a favorite is still in the works though.

### CODE REVIEW

**Comment by franklinyu** [comment]
Well, the empty migrations are created by @dbaird1 and he also messed up the schema. I never edit the schema. (I didn't even touch the database structure.) If you want the newer version of database, you can merge with his branch `bikeTests`.

**Comment by betsy-tse1** [comment]
either way, there was a :serial_num in the database when i pulled from the agileventures git repository, and this way it makes it easier to work on adding a bike to a user's favorites, since now I can see the button on the bike_list page
i could cancel my pull request if that's what you want me to do

**Comment by franklinyu** [comment]
No, there is no need to cancel this pull request. Even if there is something needed to be modified, I believe it is better to update the pull request rather than cancelling it.

**Comment by betsy-tse1** [comment]
What exactly do you want me to change?

**Comment by franklinyu** [comment]
Sorry if I got you mad.

Currently, it is really difficult to give just a 'solution', since I have no idea which is the best one. The migration you made is actually very standard and RESTful; it is how a formal migration should be, so I would like your version much better than @dbaird1's. However, he already changed his migration file for creating the `bikes` table, and deleted all the empty migrations (in 234e54e77f5d03922be50e65562e85462d02bf1b). So here is the problem: yours and his alone will work perfectly well (yours works even better), but having both just do not make sense.

In summary, I believe that you'd better talk to him about how to fix this.

**Comment by betsy-tse1** [comment]
The empty migrations are still in the most recent version of bubhub, and
the bikes table is still pretty empty. I'm not sure what happened, but the
latest commit in the db/migrate folder relating to anything that we're
talking about is 28 days ago.

I don't know where these changes are in the AgileVenture's bubhub, but when
I click to view how the file looks like after dbarid1's changes, they're
still the same empty migrations that I worked with. The history on the
master branch of bubhub doesn't show a change made by dbaird1 15 days ago,
which is the link that you provided. I have no idea what happened.

I'm not sure how to fix this because the most recent version of bubhub that
I forked on Wednesday had these issues, but you're telling me that they
were fixed somehow before Wednesday and committed and pushed to bubhub even
though I ran the most recent copy and came across errors.

**Comment by franklinyu** [comment]
Yes, the current version in Agileventure's bubhub has more than one problem. The reason why you still see the empty migration and messy schema is that @dbaird1 did not make a pull request to ask @kgoodwill to merge all his fixing commits (while he should have done this long time ago). When I was writing my story I needed the schema, but I cannot wait for @kgoodwill to pull this, so I directly pulled from @dbaird1's branch.

**Comment by betsy-tse1** [comment]
So that's great and all, but if we keep pulling from each others branches
while the original master bubhub is just left alone, we will eventually all
be on different pages and probably run into more of these issues. What I'm
trying to do is make a pull request to the master so we don't have any more
issues with adding a bike.

You needed the schema, and I needed an actual list of bikes on the
bike_list page, so I fixed it on the master branch and sent a pull request
so that there is an updated schema and actual adding function. I'm working
with the bubhub master branch. You're working with dbaird1's. We're on
completely different pages.

**Comment by betsy-tse1** [comment]
I also don't understand why there was no pull request made from dbaird1 if
he fixed migrations and created a better schema long before two days ago.

**Comment by franklinyu** [comment]
If you really want to fix this, then you may as well delete all the empty migrations so that we can know what happen by just looking at the file names. After that, I will pull from the AV-master again and solve my local merge-conflicts.

**Comment by franklinyu** [comment]
Neither do I. I guess he just forget to. He even told me that he already did that.
> I also don't understand why there was no pull request made from dbaird1 if he fixed migrations and created a better schema long before two days ago.

**Comment by franklinyu** [comment]
I closed my pull request. Please inform me if you have finished all that, and I will merge the master before I reopen that request.

**Comment by franklinyu** [comment]
Please pull from the master branch. If anything crash (possibly the `db/schema.rb` will be deleted), run this:
```bash
bundle exec rake db:drop db:create
rm db/schema.rb # double check that the old schema is deleted
bundle exec rake db:migrate
```
